### PR TITLE
Set ImageNet data augmentation by default

### PR DESCRIPTION
**Pull request by ymjiang**
https://github.com/apache/incubator-mxnet/blob/a38278ddebfcc9459d64237086cd7977ec20c70e/example/image-classification/train_imagenet.py#L42

When I try to train imagenet with this line commented, the train-accuracy reaches 99% while the validation-accuracy is only less than 50% (single machine, 8 GPUs, global batchsize=2048, Resnet50, fp32). Absolutely this is overfitting.

Then I uncomment this line and try again with the same experiment settings. This time both train and validation accuracy converge to about 66%, which looks like normal result.

Thus, it seems that this data augmentation is pretty important for ImageNet training. Perhaps it will be better to uncomment this as default, so that future developers won't get confused by the overfitting issue.

### CODE REVIEW

**Comment by Roshrini** [comment]
@sandeep-krishnamurthy @eric-haibin-lin Can you take a look?

@mxnet-label-bot Add [pr-awaiting-review]

**Comment by vishaalkapoor** [comment]
I'm unsure why image net arguments are not the default for a training script for image net and would be curious to know why, but there are two better approaches to this depending on what is determined.

If ImageNet arguments are to be the default, they should be merged into the stanza:
```
    parser.set_defaults(
        # network
        network          = 'resnet',
        num_layers       = 50,
        # data
        num_classes      = 1000,
        num_examples     = 1281167,
        image_shape      = '3,224,224',
        min_random_scale = 1, # if input image has min size k, suggest to use
                              # 256.0/x, e.g. 0.533 for 480
        # train
        num_epochs       = 80,
        lr_step_epochs   = '30,60',
        dtype            = 'float32'
    )
```
If they are not the default, it would be cleaner to add an argument --override-with-image-net-augmentations or something more appropriately named that would override parameters with those in the method.

Vishaal

**Comment by stu1130** [comment]
@rahul003 could you take a look at it. any idea why it was commented? Thanks a lot!

**Comment by sandeep-krishnamurthy** [comment]
@ymjiang - Thanks for your contributions. Did you get a chance to look at @vishaalkapoor comment?

**Comment by vandanavk** [comment]
@mxnet-label-bot update [pr-awaiting-response]

**Comment by ymjiang** [comment]
Hi @sandeep-krishnamurthy , I agree with @vishaalkapoor that the parameter argument should be set as default. But I see the parameter is already provided in  `set_imagenet_aug`. Perhaps one neat way would be to directly set it as uncommented? No other change will be involved.

**Comment by ankkhedia** [comment]
@vishaalkapoor Could you suggest a way forward on this PR?

File: example/image-classification/train_imagenet.py [hunk 1]
```diff
@@ -39,7 +39,7 @@ def set_imagenet_aug(aug):
     data.add_data_args(parser)
     data.add_data_aug_args(parser)
     # uncomment to set standard augmentations for imagenet training
```
Comment (anirudhacharya) [inline example/image-classification/train_imagenet.py]: this comment should change accordingly

**Comment by anirudhacharya** [comment]
@ymjiang can you please set a command line argument to either override or keep `set_imagenet_aug` line. I think that is what @vishaalkapoor was suggesting.

**Comment by karan6181** [comment]
@ymjiang Could you please address the review comments made by @anirudhacharya. It seems no updates since last 2 weeks. Thanks!

**Comment by ymjiang** [comment]
@karan6181 @anirudhacharya Sorry for my delay. I committed two new changes to enable data-augmentation with command-line argument. Please review if they are appropriate.

**Comment by piyushghai** [comment]
@anirudhacharya Ping for review. 
@ymjiang Can you look into the CI failures ?

**Comment by Roshrini** [comment]
@vishaalkapoor Can you take a look at this PR again?

**Comment by roywei** [comment]
@ymjiang Hi, could you rebase to latest master? it should resolve the failing CI test

**Comment by pinaraws** [comment]
@ymjiang Hi, could you rebase to latest master? it should resolve the failing CI test

**Comment by piyushghai** [comment]
@ymjiang Gentle ping...

**Comment by ymjiang** [comment]
Rebased to master now: [https://github.com/apache/incubator-mxnet/pull/15189](url). Will close this issue.
